src/Orchard.Web/Modules/Orchard.OpenId/Controllers/AccessController.cs

```diff
@@ -0,0 +1,117 @@
+        private readonly SignInManager<User> _signInManager;
+        private readonly OpenIddict.OpenIddictUserManager<User> _userManager;
+
+        public AccessController(
+            OpenIddict.OpenIddictApplicationManager<OpenIdApplication> applicationManager,
+            SignInManager<User> signInManager,
+            OpenIddict.OpenIddictUserManager<User> userManager)
+        {
+            _applicationManager = applicationManager;
+            _signInManager = signInManager;
+            _userManager = userManager;
+        }
+
+        [Authorize, HttpGet]
+        public async Task<IActionResult> Authorize()
+        {
+            var request = HttpContext.GetOpenIdConnectRequest();
+
+            var application = await _applicationManager.FindByClientIdAsync(request.ClientId);
+            if (application == null)
+            {
+                return View("Error", new ErrorViewModel
+                {
+                    Error = OpenIdConnectConstants.Errors.InvalidClient,
+                    ErrorDescription = "Details concerning the calling client application cannot be found in the database"
+                });
+            }
+
+            if (application.SkipConsent)
+                return await IssueAccessIdentityTokens();
+
+            return View(new AuthorizeViewModel
+            {
+                ApplicationName = application.DisplayName,
+                RequestId = request.RequestId,
+                Scope = request.Scope
+            });
+        }
+
+        [Authorize, HttpPost, ValidateAntiForgeryToken]
+        public Task<IActionResult> Accept()
+        {
+            return IssueAccessIdentityTokens();
+        }
+
+        private async Task<IActionResult> IssueAccessIdentityTokens()
+        {
+            // Extract the authorization request from the ASP.NET environment.
+            var request = HttpContext.GetOpenIdConnectRequest();
+
+            // Retrieve the profile of the logged in user.
+            var user = await _userManager.GetUserAsync(User);
+            if (user == null)
+            {
+                return View("Error", new ErrorViewModel
+                {
+                    Error = OpenIdConnectConstants.Errors.ServerError,
+                    ErrorDescription = "An internal error has occurred"
+                });
+            }
+
+            // Create a new ClaimsIdentity containing the claims that
+            // will be used to create an id_token, a token or a code.
+            var identity = await _userManager.CreateIdentityAsync(user, request.GetScopes());
+
+            // Create a new authentication ticket holding the user identity.
+            var ticket = new AuthenticationTicket(
+                new ClaimsPrincipal(identity),
+                new AuthenticationProperties(),
+                OpenIdConnectServerDefaults.AuthenticationScheme);
+
+            ticket.SetResources(request.GetResources());
```

Comment:
Nice.

In the meantime, it's important to note that enabling the client credentials has side effects. Orchard's controllers/authorization policies will likely have to be updated to distinguish tokens representing users and tokens representing client applications.

Related ticket: https://github.com/openiddict/openiddict-core/issues/138